prometheus-xmpp-alerts

```diff
@@ -157,15 +172,25 @@ async def serve_alert(request):
         alert = await request.json()
     except json.decoder.JSONDecodeError as e:
         raise web.HTTPUnprocessableEntity(str(e))
-    if 'format' in config and config['format'] == 'full':
-        text = '\n--\n'.join(create_message_full(alert))
-    else:
-        text = '\n'.join(create_message_short(alert))
-    id_ = app.send_message(
-            mto=to_jid,
-            mbody=text,
-            mtype='chat')
-    return web.Response(body='Sent message')
+    try:
+        if 'format' in config and config['format'] == 'full':
+            text = '\n--\n'.join(create_message_full(alert))
+        else:
+            text = '\n'.join(create_message_short(alert))
+        try:
+            id_ = app.send_message(
+                    mto=to_jid,
+                    mbody=text,
+                    mtype='chat')
+            last_alert_message_succeeded_gauge.set(1)
+            return web.Response(body='Sent message')
+        except slixmpp.xmlstream.xmlstream.NotConnectedError as e:
```

Comment:
same here - when would web.Response raise a NotConnectedError?

Reply:
This does raise that error when `app.send_message()` fails because you lost connectivity in the interim.  I've had this happen to me, hence why there is a gauge set zero in the catch.